src/engine/QueryPlanner.h

```diff
@@ -201,6 +201,18 @@ class QueryPlanner {
   SubtreePlan getTextLeafPlan(const TripleGraph::Node& node) const;
 
   SubtreePlan optionalJoin(const SubtreePlan& a, const SubtreePlan& b) const;
+
+  /**
+   * @brief Determines if the pattern trick (and in turn the
+   * CountAvailablePredicates operation) are applicable to the given
+   * parsed query. If a has predicate triple is found and
+   * CountAvailblePredicates can be used for it the triple is removed from the
```

Comment:
"for it, the"

Also can you add the information from your comment that if the check fails the `ql:has-predicate` part of the query will still be computed just without the pattern trick

Reply:
I added the comment that the query will still be computed to the function call instead, as it is not directly a part of the `checkUsePatternTrick` function.